django/utils/text.py

```diff
@@ -393,6 +393,22 @@ def compress_sequence(sequence, *, max_random_bytes=None):
     yield buf.read()
 
 
+async def acompress_sequence(sequence, *, max_random_bytes=None):
+    buf = StreamingBuffer()
+    filename = _get_random_filename(max_random_bytes) if max_random_bytes else None
+    with GzipFile(
+        filename=filename, mode="wb", compresslevel=6, fileobj=buf, mtime=0
+    ) as zfile:
+        # Output headers...
+        yield buf.read()
+        async for item in sequence:
+            zfile.write(item)
+            data = buf.read()
```

Comment:
Is this blocked on ticket-36293? Without flushing here, your sample app from the ticket blocks. Or does this have independent value?

Reply:
This change has independent value because it fixes response correctness, even if they are buffered.